File: common/src/main/java/com/pluxity/feature/dto/FeatureResponse.java
package com.pluxity.feature.dto;

import com.pluxity.feature.entity.Feature;
import com.pluxity.feature.entity.Spatial;

public record FeatureResponse(
        String id, Spatial position, Spatial rotation, Spatial scale, Long assetId, String floorId) {

    public static FeatureResponse from(Feature feature) {
        return new FeatureResponse(
                feature.getId(),
                feature.getPosition(),
                feature.getRotation(),
                feature.getScale(),
                feature.getAssetId(),
                feature.getFloorId());
    }
}

File: common/src/main/java/com/pluxity/feature/dto/FeatureResponse.kt
package com.pluxity.feature.dto

import com.pluxity.feature.entity.Feature
import com.pluxity.feature.entity.Spatial

data class FeatureResponse(
    val id: String,
    val position: Spatial?,
    val rotation: Spatial?,
    val scale: Spatial?,
    val assetId: Long?,
    val floorId: String?,
)

fun Feature.toFeatureResponse() =
    FeatureResponse(
        id = this.id!!,
        position = this.position,
        rotation = this.rotation,
        scale = this.scale,
        assetId = this.assetId,
        floorId = this.floorId,
    )
